src/supa/documents/topology.py

```diff
@@ -0,0 +1,153 @@
+
+            topology = Element(QName(nsmap["topo"], "Topology"), nsmap=nsmap)
+            topology.set("id", network_id)
+            topology.set("version", now.isoformat(timespec="seconds"))
+            name = SubElement(topology, "name")
+            name.text = settings.topology_name
+            lifetime = SubElement(topology, "Lifetime")
+            lifetime_start = SubElement(lifetime, "start")
+            lifetime_start.text = now.isoformat(timespec="seconds")
+            lifetime_end = SubElement(lifetime, "end")
+            lifetime_end.text = (now + timedelta(weeks=1)).isoformat(timespec="seconds")
+            service_definition = SubElement(topology, QName(nsmap["sd"], "serviceDefinition"))
+            service_definition.set("id", f"{network_id}:sd:EVTS.A-GOLE")
+            service_definition_name = SubElement(service_definition, "name")
+            service_definition_name.text = "GLIF Automated GOLE Ethernet VLAN Transfer Service"
+            service_definition_service_type = SubElement(service_definition, "serviceType")
+            service_definition_service_type.text = "http://services.ogf.org/nsi/2013/12/descriptions/EVTS.A-GOLE"
+            relation = SubElement(topology, "Relation")
+            relation.set("type", "http://schemas.ogf.org/nml/2013/05/base#hasService")
+            relation_switching_service = SubElement(relation, "SwitchingService")
+            relation_switching_service.set("id", "urn:ogf:network:surf.nl:2020:production:switch:EVTS.A-GOLE")
+            relation_switching_service.set("labelSwapping", "true")
+            relation_switching_service.set("labelType", "http://schemas.ogf.org/nml/2012/10/ethernet#vlan")
+            relation_switching_service_definition = SubElement(
+                relation_switching_service, QName(nsmap["sd"], "serviceDefinition")
+            )
+            relation_switching_service_definition.set("id", "urn:ogf:network:surf.nl:2020:production:sd:EVTS.A-GOLE")
+            for stp in stps:
+                bidirectional_port = SubElement(topology, "BidirectionalPort")
+                bidirectional_port.set("id", f"{network_id}:{stp.stp_id}")
+                bidirectional_port_name = SubElement(bidirectional_port, "name")
+                bidirectional_port_name.text = stp.description
+                bidirectional_port_group = SubElement(bidirectional_port, "PortGroup")
+                bidirectional_port_group.set("id", f"{network_id}:{stp.stp_id}:in")
+                bidirectional_port_group = SubElement(bidirectional_port, "PortGroup")
+                bidirectional_port_group.set("id", f"{network_id}:{stp.stp_id}:out")
+            relation = SubElement(topology, "Relation")
+            relation.set("type", "http://schemas.ogf.org/nml/2013/05/base#hasInboundPort")
+            for stp in stps:
+                relation_port_group = SubElement(relation, "PortGroup")
+                relation_port_group.set("id", f"{network_id}:{stp.stp_id}:in")
```

Comment:
lot's of meaning in the id's (not a real problem -> but conventions in these kind of situations are very hard to change....)

Reply:
Luckily this has been standardised and documented in https://www.ogf.org/documents/GFD.206.pdf.